### PR TITLE
use google-github-actions/auth in pre-release check

### PR DESCRIPTION
**Pull request by clairemcginty**
Switches to https://github.com/google-github-actions/auth with a GHA-managed secret rather than manually decrypting a service account key. We can switch the other steps to use this too.

proof it works!: https://github.com/spotify/scio/runs/4732201000

### CODE REVIEW

File: .github/workflows/pre-release-check.yml [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
+  DEFAULT_DATAFLOW_ARGS: --runner=DataflowRunner --project=data-integration-test --region=europe-west1 --tempLocation=gs://dataflow-tmp-europe-west1/gha
 
 on:
   workflow_dispatch # Manually triggered
 jobs:
   test-dataflow:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
```
Comment (regadas) [inline .github/workflows/pre-release-check.yml:14]: it seems that if we use SA key setting permissions is not required?

**Review by regadas — COMMENTED** [review]
I like this! and yeah we should then update the other workflows

File: .github/workflows/pre-release-check.yml (hunk 1 above)
Comment (clairemcginty) [reply .github/workflows/pre-release-check.yml]: you're right! thanks 👍

**Comment by clairemcginty** [comment]
@regadas thanks, updated the other workflows!

**Pull request merged**